### PR TITLE
ci: scroll syncing

### LINKED ISSUE #37: Reth vs L2 Geth Integration Consistency Tests
# Overview
We should develop a pattern to perform consistency checks against L2 Geth to ensure correctness.

# Design Considerations
L2 geth does not support the engine API and as such we can not use the engine API to drive block production of L2 geth.

## Peter Prosal
Generate a chain of test blocks using l2 geth and feed them to reth and assert consistent state root. We can serialize these blocks and run this integration test via CI.

### PR DESCRIPTION
**Pull request by greged93**
Adds a node syncing of Scroll mainnet up to block 50000 on pushes to main and scroll branches in CI.

Resolves #37

### CODE REVIEW

**Comment by frisitano** [comment]
Can you merge the `scroll` branch please?

File: .github/workflows/sync.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ name: sync test
 
 on:
   merge_group:
+  push:
+    branches: [main, scroll]
```
Comment (frisitano) [inline .github/workflows/sync.yml:8]: Currently, we don't have a merge queue so I believe the first event `merge_group` will never trigger and we will not do any pre-merge validation of sync. This is probably fine for now the post-merge `push` trigger should be sufficient but I will add an issue to introduce a merge queue as the repo begins to stabilize and we are moving into production readiness.

File: .github/workflows/sync.yml [hunk 2]
```diff
@@ -35,6 +37,12 @@ jobs:
             tip: "0xbb9b85352c7ebca6ba8efc63bd66cecd038c92ec8ebd02e153a3e0b197e672b7"
             block: 10000
             unwind-target: "0x118a6e922a8c6cab221fc5adfe5056d2b72d58c6580e9c5629de55299e2cf8de"
+          - build: install-scroll
+            bin: scroll-reth
+            chain: scroll-mainnet
+            tip: "0x1f398ce1e03b9d7d7fcba8512dc43c9f84ecaffb15954ab178fab48151e84484"
+            block: 50000
```
Comment (frisitano) [inline .github/workflows/sync.yml:44]: Did you benchmark how long it takes to run 50000 blocks?

**Review by frisitano — APPROVED** [review]
Looks good! LGTM. One comment and question inline.

File: .github/workflows/sync.yml (hunk 2 above)
Comment (greged93) [reply .github/workflows/sync.yml:44]: checking now

Comment (greged93) [reply .github/workflows/sync.yml:44]: took 702.25 seconds to run

Comment (frisitano) [reply .github/workflows/sync.yml:44]: Ok thanks for looking into this! This should be fine as it'll only run on merge queue and on merge to main/scroll branches.

**Pull request merged**